### PR TITLE
Core: Deprecate negative epoch timestamps

### PR DESCRIPTION
**Pull request by rjernst**
Negative timestamps are currently supported in joda time. These are
dates before epoch. However, it doesn't really make sense to have a
negative timestamp, since this is a modern format. Any dates before
epoch can be represented with normal date formats, like ISO8601.
Additionally, implementing negative epoch timestamp parsing in java time
has an edge case which would more than double the code required. This
commit deprecates use of negative epoch timestamps.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/common/joda/SimpleJodaTests.java [hunk 1]
```diff
@@ -262,49 +262,6 @@ public void testThatEpochsCanBeParsed() {
         assertThat(formatter.parseJoda(epochFloatValue).getMillis(), is(dateTime.getMillis()));
     }
 
-    public void testThatNegativeEpochsCanBeParsed() {
```
Comment (jasontedor) [inline server/src/test/java/org/elasticsearch/common/joda/SimpleJodaTests.java]: Why is this test removed? These would still be parsed by Joda, so would issue deprecation warnings? Or am I misunderstanding something?

**Review by jasontedor — COMMENTED** [review]
I left a question.

File: server/src/test/java/org/elasticsearch/common/joda/SimpleJodaTests.java (hunk 1 above)
Comment (rjernst) [reply server/src/test/java/org/elasticsearch/common/joda/SimpleJodaTests.java]: I added it back with a warning check in efd290b

**Comment by rjernst** [comment]
@elasticmachine test this please

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by rjernst** [comment]
@elasticmachine test this please

**Pull request merged**